handlers/authentication.go

```diff
@@ -383,6 +440,8 @@ func (aHandler AuthenticationHandler) Handle(next http.Handler) http.Handler {
 		case config.AuthStrategyLogin:
 			statusCode = checkKialiSession(w, r)
 			token = aHandler.saToken
+		case config.AuthStrategyToken:
+			statusCode, token = checkOpenshiftSession(w, r)
```

Comment:
Is it always Openshift?

Reply:
`checkOpenshiftSession` works OK.
But I created a dedicated function, per my comment in `performTokenAuthentication` (`In Kubernetes, the version API seems to allow anonymous access, so it's not feasible to use the version API for token verification`)

Good catch.